bin/metaBackend.py

```diff
@@ -21,47 +21,506 @@
+            for c in t["columns"]:
+                if c["name"] == cName:
+                    foundColumn = True
+                    c["ord_pos"] = int(ordP)
+                    break
+            if not foundColumn:
+                self._log.error(
+                    "Column '%s.%s' not found in db, present in ascii file.",
+                    tName, cName)
+                raise MetaBException(MetaBException.COL_NOT_IN_TB, cName, tName)
+
+        # Check if we covered all columns
+        for t in theTable:
+            for c in theTable[t]["columns"]:
+                if "ord_pos" not in c:
+                    msg = "Column '%s.%s' not found in " % (tName, cName)
+                    msg += "ascii file, present in db."
+                    self._log.error(msg)
+                    raise MetaBException(MetaBException.COL_NOT_IN_FL, cName, tName)
+
+        # Get schema description and version, it is ok if it is missing
+        ret = db.execCommand1(
+            "SELECT version, descr FROM %s.ZZZ_Schema_Description" % dbName)
+        if not ret:
+            self._log.error(
+                "Db '%s' does not contain schema version/description", dbName)
+            schemaVersion = "unknown"
+            schemaDescr = ""
+        else:
+            (schemaVersion, schemaDescr) = ret
+
+        # This can be sometimes handy for debugging
+        # pp = pprint.PrettyPrinter(indent=2)
+        # pp.pprint(theTable)
+
+        # Get host/port from authFile
+        dict = readCredentialFile(dbMysqlAuthF, self._log)
+        (host, port) = [dict[k] for k in ('host', 'port')]
+
+        # Now, we will be talking to the metaserv database, so change
+        # connection as needed
+        if self._msMysqlAuthF != dbMysqlAuthF:
+            db = Db(read_default_file=self._msMysqlAuthF)
+
+        # get ownerId, this serves as validation that this is a valid owner name
+        ret = db.execCommand1("SELECT userId FROM User WHERE mysqlUserName = %s",
+                              (owner,))
+        if not ret:
+            self._log.error("Owner '%s' not found.", owner)
+            raise MetaBException(MetaBException.OWNER_NOT_FOUND, owner)
+        ownerId = ret[0]
+
+        # get projectId, this serves as validation that this is a valid project name
+        ret = db.execCommand1("SELECT projectId FROM Project WHERE projectName =%s",
+                              (projectName,))
+        if not ret:
+            self._log.error("Project '%s' not found.", owner)
+            raise MetaBException(MetaBException.PROJECT_NOT_FOUND, projectName)
+        projectId = ret[0]
+
+        # Finally, save things in the MetaServ database
+        cmd = "INSERT INTO Repo(url, projectId, repoType, lsstLevel, dataRelease, "
+        cmd += "version, shortName, description, ownerId) "
+        cmd += "VALUES('/dummy',%s,'db',%s,%s,%s,%s,%s,%s) "
+        opts = (str(projectId), level, dataRel, schemaVersion, dbName, schemaDescr,
```

Comment:
The error comes from a debug log message that is explicitly joining the parameters assuming they are all strings.  The conversion should be applied there -- something like simply using `optParams` to replace the whole join would be simplest, letting the string formatter do the conversion -- not in the caller.